src/app/store/effects.ts

```diff
@@ -1,28 +1,28 @@
 import { Injectable } from '@angular/core';
 import { Actions, createEffect, ofType } from '@ngrx/effects';
 import {
-  getLogin,
-  getLoginComplete,
-  getLogout,
-  getLogoutComplete,
-  getRegister,
-  getRegisterComplete,
+  getBooks,
+  getBooksComplete,
+  getBookshelf,
+  getBookshelfComplete,
 } from './actions';
 import { AuthService } from '../services/auth.service';
 import { catchError, EMPTY, map, retry, switchMap } from 'rxjs';
+import { DataService } from '../services/data.service';
 
 @Injectable()
 export class UserEffects {
-  getlogin$ = createEffect(() =>
+  getBookshelf$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(getLogin.type),
-      switchMap((action: { email: string; password: string }) =>
-        this.authService.login(action.email, action.password).pipe(
-          map(({ name, id }) => getLoginComplete({ name, id })),
-          retry(1),
+      ofType(getBookshelf.type),
+      switchMap((action: { userId: string }) =>
+        this.dataService.getBookshelf(action.userId).pipe(
+          map((response) =>
+            getBookshelfComplete({ bookshelfId: response[0].id.toString() }),
+          ),
           catchError((error) => {
             console.error(
-              'An unexpected error occurred while trying to log you in: ',
+              'An unexpected error occurred while trying to get your bookshelf: ',
```

Comment:
Maybe add a method that the user can easily see this error. Users do not really check console logs

Reply:
Proper user feedback will be added in final UI fixes and updates when finalising the project.